pkg/controller/replicaset/replica_set.go

```diff
@@ -193,6 +194,44 @@ func (rsc *ReplicaSetController) Run(workers int, stopCh <-chan struct{}) {
 	<-stopCh
 }
 
+// getReplicaSetsWithSameController returns a list of ReplicaSets with the same
+// owner as the given ReplicaSet.
+func (rsc *ReplicaSetController) getReplicaSetsWithSameController(rs *apps.ReplicaSet) []*apps.ReplicaSet {
+	controllerRef := metav1.GetControllerOf(rs)
+	if controllerRef == nil {
+		utilruntime.HandleError(fmt.Errorf("ReplicaSet has no controller: %v", rs))
+		return nil
+	}
+
+	allRSs, err := rsc.rsLister.ReplicaSets(rs.Namespace).List(labels.Everything())
+	if err != nil {
+		utilruntime.HandleError(err)
+		return nil
+	}
+
+	var relatedRSs []*apps.ReplicaSet
+	for _, r := range allRSs {
+		if ref := metav1.GetControllerOf(r); ref != nil && ref.UID == controllerRef.UID {
+			relatedRSs = append(relatedRSs, r)
+		}
+	}
+
+	if len(relatedRSs) == 0 {
+		// Note: This is impossible because at least rs should have the
+		// same controller as itself.
+		utilruntime.HandleError(fmt.Errorf("no ReplicaSet has controller %v", controllerRef))
```

Comment:
You'll hit this case if there is a pending delete in the queue, so I'm not sure you should be `utilruntime.HandleError` here.  This is really v(4)

Reply:
Are you sure?  I believe the `rsc.rsLister.ReplicaSets(rs.Namespace).List(labels.Everything())` above should still list the ReplicaSet even if it is pending deletion.  This test case passes: https://github.com/kubernetes/kubernetes/pull/80004/commits/de3f4955c09a7159974ccb8479e8edbd87c009e9#diff-16e152a1acd1373ada706d8cb4c099f6R374-R379

Reply:
it could already have been deleted in the cache, and the queue can have the delete event, but you could be processing the last edit.